framework/src/kernels/NodalScalarKernel.C

```diff
@@ -12,27 +12,56 @@
 // MOOSE includes
 #include "Assembly.h"
 #include "MooseVariableScalar.h"
+#include "MooseMesh.h"
 #include "SystemBase.h"
 
 template <>
 InputParameters
 validParams<NodalScalarKernel>()
 {
   InputParameters params = validParams<ScalarKernel>();
-  params.addRequiredParam<std::vector<dof_id_type>>("nodes", "Node ids");
+  params.addParam<std::vector<dof_id_type>>("nodes", "Supply nodes using node ids");
+  params.addParam<std::vector<BoundaryName>>("nodeset_names", "Suppy nodes using named boundaries");
+
   return params;
 }
 
 NodalScalarKernel::NodalScalarKernel(const InputParameters & parameters)
   : ScalarKernel(parameters),
     Coupleable(this, true),
     MooseVariableDependencyInterface(),
-    _node_ids(getParam<std::vector<dof_id_type>>("nodes"))
+    _node_ids(getParam<std::vector<dof_id_type>>("nodes")),
+    _node_bc_names(getParam<std::vector<BoundaryName>>("nodeset_names"))
 {
   // Fill in the MooseVariable dependencies
   const std::vector<MooseVariable *> & coupled_vars = getCoupledMooseVars();
   for (const auto & var : coupled_vars)
     addMooseVariableDependency(var);
+
+  // Check if node_ids and/or node_bc_names given
+  if ((_node_ids.size() == 0) && (_node_bc_names.size() == 0))
+    mooseError("Must provide one of nodes or node_names");
+
+  if ((_node_ids.size() != 0) && (_node_bc_names.size() != 0))
+    mooseError("Must provide only one input using nodes or node_names");
+
+  // nodal bc names provided, append the nodes in each bc to _node_ids
+  if ((_node_ids.size() == 0) && (_node_bc_names.size() != 0))
+  {
+    std::vector<dof_id_type> nodelist;
+
+    for (auto i = 0; i < _node_bc_names.size(); i++)
```

Comment:
`std::size` instead of `auto`

Reply:
More information on this. `auto` is fine and encouraged but in this usage it's actually incorrect! Literal numbers actually give you back a signed int which will give you sign mismatch warnings.

If you really like auto, we created a macro that will do the right thing.
```
auto i = beginIndex(_node_bc_names);  ...
```